### PR TITLE
Bump `thiserror` to `1.0.40` and `jni` to `0.21.1`

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

- [ ] Does this PR affect the User ?

### CODE REVIEW

**Review by touilleMan — COMMENTED** [review]
No change in cargo.lock ?

**Review by FirelightFlagboy — COMMENTED** [review]
Could you complete the PR description why you need to udpate `thiserror` before the next month (dependabot auto-update)?

File: oxidation/bindings/android/libparsec/rust/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ crate-type = ["staticlib", "cdylib"]
 [dependencies]
 libparsec = { path = "../../../../libparsec" }
 lazy_static = "1.4.0"
-jni = "0.20.0"
+jni = "0.21.1"
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply oxidation/bindings/android/libparsec/rust/Cargo.toml:12]: `jni` depends on `thiserror`

**Comment by TimeEngineer** [comment]
> No change in cargo.lock ?

What ?

File: oxidation/bindings/android/libparsec/rust/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply oxidation/bindings/android/libparsec/rust/Cargo.toml:12]: > Could you complete the PR description why you need to udpate `thiserror` before the next month (dependabot auto-update)?

This is not needed, but I work on winfsp so I had this change, so why not ?

**Pull request merged**